From: Aristeu Rozanski <aris@redhat.com>
Date: Wed, 20 Aug 2008 13:16:59 -0400
Subject: [x86_64] nmi: add missing prototypes in xen headers
Message-id: 20080820171649.605271000@redhat.com
O-Subject: [RHEL5.3 PATCH 13/25] nmi: add missing prototypes in Xen headers
Bugzilla: 447618

https://bugzilla.redhat.com/show_bug.cgi?id=447618

on RHEL-5, Xen has a different header for NMI prototypes and it needs to be
updated to include the new functions.

upstream: not relevant upstream, Xen shares the common header

diff --git a/include/asm-x86_64/mach-xen/asm/nmi.h b/include/asm-x86_64/mach-xen/asm/nmi.h
index 8fff33f..3c75095 100644
--- a/include/asm-x86_64/mach-xen/asm/nmi.h
+++ b/include/asm-x86_64/mach-xen/asm/nmi.h
@@ -71,6 +71,14 @@ static inline unsigned char get_nmi_reason(void)
 extern int panic_on_timeout;
 extern int unknown_nmi_panic;
 
+void lapic_watchdog_stop(void);
+int lapic_watchdog_probe(void);
+int lapic_watchdog_init(unsigned nmi_hz);
+int lapic_wd_event(unsigned nmi_hz);
+unsigned lapic_adjust_nmi_hz(unsigned hz);
+void disable_lapic_nmi_watchdog(void);
+void enable_lapic_nmi_watchdog(void);
+
 extern int check_nmi_watchdog(void);
  
 extern void setup_apic_nmi_watchdog (void);
@@ -78,7 +86,7 @@ extern int reserve_lapic_nmi(void);
 extern void release_lapic_nmi(void);
 extern void disable_timer_nmi_watchdog(void);
 extern void enable_timer_nmi_watchdog(void);
-extern void nmi_watchdog_tick (struct pt_regs * regs, unsigned reason);
+extern int nmi_watchdog_tick (struct pt_regs * regs, unsigned reason);
 
 extern void nmi_watchdog_default(void);
 extern int setup_nmi_watchdog(char *);
